jvcl/run/JvMemoryDataset.pas

```diff
@@ -111,49 +111,61 @@   TJvMemoryRecord = class;
   TJvRecordBuffer = Pointer;
   {$ENDIF RTL240_UP}
 
-  {$IFDEF RTL230_UP}
-  [ComponentPlatformsAttribute(pidWin32 or pidWin64)]
-  {$ENDIF RTL230_UP}
-  TJvMemoryData = class(TDataSet)
-  private
+  TJvMemDataCloneOption = (coWithEvents, coWithFieldEvents, coClosed);
+  TJvMemDataCloneOptions = set of TJvMemDataCloneOption;
+
+  PJvMemDataCommon = ^TJvMemDataCommon;
+  TJvMemDataCommon = record
```

Comment:
I'm sorry, but I don't see the point of this record, and it gets even worse with its allocation being done manually via a `getmem`.